charts/consul/templates/client-snapshot-agent-deployment.yaml

```diff
@@ -1,4 +1,5 @@
 {{- if (or (and (ne (.Values.client.enabled | toString) "-") .Values.client.enabled) (and (eq (.Values.client.enabled | toString) "-") .Values.global.enabled)) }}
+{{- if or (and .Values.client.snapshotAgent.configSecret.secretName (not .Values.client.snapshotAgent.configSecret.secretKey)) (and (not .Values.client.snapshotAgent.configSecret.secretName) .Values.client.snapshotAgent.configSecret.secretKey) }}{{fail "client.snapshotAgent.configSecret.secretKey and secretName must both be specified." }}{{ end -}}
```

Comment:
Nit. I would change the fail message to be name/key and a little more specific to:

`fail "client.snapshotAgent.configSecret.secretName and client.snapshotAgent.configSecret.secretKey must both be specified."`